metroid/publish.py

```diff
@@ -51,11 +51,38 @@ def publish_event(
         'subject': subject,
     }
     logger.info('Posting event to Metro topic %s. Data: %s', topic_name, formatted_data)
-    metro_response = requests.post(
-        url=f'https://api.intility.no/metro/{topic_name}',
-        headers={'content-type': 'application/json', 'x-metro-key': settings.get_x_metro_key(topic_name=topic_name)},
-        data=json.dumps(formatted_data),
-    )
-    logger.info('Posted to metro')
-    metro_response.raise_for_status()
+
+    try:
+
+        metro_response = requests.post(
+            url=f'https://api.intility.no/metro/{topic_name}',
+            headers={
+                'content-type': 'application/json',
+                'x-metro-key': settings.get_x_metro_key(topic_name=topic_name),
+            },
+            data=json.dumps(formatted_data),
+        )
+        metro_response.raise_for_status()
+        logger.info('Posted to metro')
+    except Exception as error:
+        from metroid.models import FailedPublishMessage
```

Comment:
I see I've done this myself, but there's not really any reason to have this inside the functions, so just move it to the top

Reply:
When we import at the top we get "AppRegistryNotReady: Apps aren't loaded yet."